UI/TraceDataLinux.cpp

```diff
@@ -0,0 +1,316 @@
+      const int maxFrameIndex = std::min(frameIndex + numFramesPerStep, moduleFrameList[i].m_Frames.count()); 
+
+      for ( int j = frameIndex; j < maxFrameIndex; j++ )
+      {
+        command << (moduleFrameList[i].m_Frames[j] - imageBase) << " ";
+      }
+      FILE* fd = popen(command.str().c_str(), "r");
+  
+      if ( fd == nullptr )
+      {
+        qDebug() << "Cannot resolve symbols. addr2line not found.";
+        pclose(fd);
+        ResolveResult r;
+        return r;
+      }
+
+      char* symbolBuffer = nullptr;
+      size_t symbolBufferSize;
+      while ( getline(&symbolBuffer, &symbolBufferSize, fd) != -1 )
+      {
+        if ( strcmp(symbolBuffer, "??\n") == 0 )
+        {
+          // Attempt to use objdump to get the data
+          if ( !ResolveSymbolViaObjdump(&symbolBuffer, &symbolBufferSize, hdr->GetModuleName(*moduleFrameList[i].m_Entry), moduleFrameList[i].m_Frames[frameIndex] - moduleFrameList[i].m_Entry->m_ImageBase))
+          {
+            int needed = snprintf(symbolBuffer, symbolBufferSize, "[%016lx in %s]", moduleFrameList[i].m_Frames[frameIndex], symbolFilename.str().c_str());
+            if ( needed >= symbolBufferSize )
+            {
+              symbolBufferSize = needed + 1;
+              symbolBuffer = (char*)realloc(symbolBuffer, symbolBufferSize);
+              snprintf(symbolBuffer, symbolBufferSize, "[%016lx in %s]", moduleFrameList[i].m_Frames[frameIndex], symbolFilename.str().c_str());
+            }
+          }
+        }
+
+        char* lineBuffer = nullptr;
+        size_t length;
+        getline(&lineBuffer, &length, fd);
+        
+        char* filename = nullptr;
+        int lineNumber;   
+        int elementsFilled = sscanf(lineBuffer, "%m[^:]:%d %*s", &filename, &lineNumber);
+        free(lineBuffer);
+        if ( elementsFilled == 0 )
+        {
+          qDebug() << "Unexpected response from addr2line. Aborting.";
+          ResolveResult r;
+          pclose(fd);
+          return r;
+        }
+        else if ( elementsFilled == 1 )
+        {
+          lineNumber = -1;
+        }
+  
+        SerializedSymbol out_sym = { moduleFrameList[i].m_Frames[frameIndex] };
+        out_sym.m_SymbolName = intern_string(symbolBuffer);
+        out_sym.m_FileName = intern_string(filename);
+        out_sym.m_LineNumber = lineNumber;
+        out_sym.m_ModuleIndex = i;
+        out_sym.m_Displacement = -1;
+        symbols.push_back(out_sym);
+        ++frameIndex;
+        ++resolve_count;
+        ++completed;
+        free(filename);
+        if (0 == (completed & 0x400))
+        {
+          Q_EMIT symbolResolutionProgressed(completed, total);
+        }
+      }
+      free(symbolBuffer);    
+      pclose(fd);
+    } while ( frameIndex < moduleFrameList[i].m_Frames.count() );
+  }
+
+  std::sort(symbols.begin(), symbols.end(), [](const SerializedSymbol& l, const SerializedSymbol& r) -> bool
```

Comment:
It's unfortunate we now have two copies of the basic symbol resolving logic mixed in with platform-specific resolving code. Suggest we would have a "resolver" class that implements the specifics of module loading and symbol resolution and let the actual file format code be shared and platform-independent. That way maintenance is way less likely to break e.g. Linux.